parser/parser.go

```diff
@@ -192,26 +217,52 @@ func ParseSelect(statement *sqlparser.Select) (logical.Node, error) {
 
 	// Now we only keep the selected variables.
 	if len(statement.SelectExprs) >= 1 {
-		if _, ok := statement.SelectExprs[0].(*sqlparser.StarExpr); !ok {
-			nameExpressions := make([]logical.NamedExpression, len(statement.SelectExprs))
-			for i := range expressions {
-				if !aggregating {
-					nameExpressions[i] = logical.NewVariable(expressions[i].Name())
+		nameExpressions := make([]logical.NamedExpression, len(nonStarExpressions))
+
+		// Handle non star expressions
+		for i := range nonStarExpressions {
+			if !aggregating {
+				nameExpressions[i] = logical.NewVariable(expressions[i].Name())
+			} else {
+				if len(aggregatesAs[i]) > 0 {
+					nameExpressions[i] = logical.NewVariable(aggregatesAs[i])
 				} else {
-					if len(aggregatesAs[i]) > 0 {
-						nameExpressions[i] = logical.NewVariable(aggregatesAs[i])
+					if !aggregateStars[i] {
+						nameExpressions[i] = logical.NewVariable(octosql.NewVariableName(fmt.Sprintf("%v_%v", expressions[i].Name(), aggregates[i])))
 					} else {
-						if !aggregateStars[i] {
-							nameExpressions[i] = logical.NewVariable(octosql.NewVariableName(fmt.Sprintf("%v_%v", expressions[i].Name(), aggregates[i])))
-						} else {
-							nameExpressions[i] = logical.NewVariable(octosql.NewVariableName(fmt.Sprintf("%v_%v", "*star*", aggregates[i])))
-						}
+						nameExpressions[i] = logical.NewVariable(octosql.NewVariableName(fmt.Sprintf("%v_%v", octosql.StarExpressionName, aggregates[i])))
 					}
 				}
 			}
 
-			root = logical.NewMap(nameExpressions, root, false)
+			// We check for duplicated columns
+			for j := 0; j < i; j++ {
+				if nameExpressions[j].Name() == nameExpressions[i].Name() {
+					return nil, errors.New("select expressions contain duplicated columns")
+				}
+			}
 		}
+
+		// Handle star expressions collisions
+		for i := len(nonStarExpressions); i < len(statement.SelectExprs); i++ {
+			qualifier := expressions[i].Name().Source()
+
+			if qualifier == "" { // a * expression collides with anything else
+				if len(statement.SelectExprs) > 1 {
```

Comment:
no! it doesn't actually.
this is bad: SELECT *, p.name FROM people p
this isn't: SELECT *, p.name as mename FROM people p
or even: SELECT test, (SELECT *, p.test FROM people g) FROM people p

Though we can ignore that for now and leave a TODO here, as you can easily work around it using qualified star expressions.

Reply:
Oh you are absolutely right! But to be honest I don't see a simple fix for this rn, because at the time of parsing the query you have no way of telling what the column names are gonna be. Leaving as TODO